luda-editor/client/src/pages/sequence_edit_page/components/image_picker/mod.rs

```diff
@@ -0,0 +1,138 @@
+use crate::{
+    color,
+    pages::sequence_edit_page::atom::{IMAGES_ATOM, SEQUENCE_ATOM},
+    storage::get_project_image_url,
+};
+use namui::prelude::*;
+use namui_prebuilt::{table::hooks::TableCell, *};
+use rpc::data::{Cut, ImageWithLabels};
+
+const OUTER_PADDING: Px = px(8.0);
+const INNER_PADDING: Px = px(4.0);
+const THUMBNAIL_WH: Wh<Px> = Wh {
+    width: px(144.0),
+    height: px(144.0),
+};
+
+#[namui::component]
+pub struct ImagePicker<'a> {
+    pub wh: Wh<Px>,
+    pub project_id: Uuid,
+    pub cut: &'a Cut,
+    pub on_event: &'a dyn Fn(Event),
+}
+
+pub enum Event {
+    Close,
+}
+pub enum InternalEvent {
+    ImageThumbnailClicked { image_id: Uuid },
+}
+
+impl Component for ImagePicker<'_> {
+    fn render(self, ctx: &RenderCtx) -> RenderDone {
+        let Self {
+            wh,
+            project_id,
+            cut,
+            on_event,
+        } = self;
+        let (images, _set_images) = ctx.atom(&IMAGES_ATOM);
+
+        let on_internal_event = |event: InternalEvent| match event {
+            InternalEvent::ImageThumbnailClicked { image_id } => {
+                add_image_to_cut(image_id, cut.id);
+                on_event(Event::Close);
+            }
+        };
+
+        let max_items_per_row = ctx.memo(|| (wh.width / (THUMBNAIL_WH.width)).floor() as usize);
+        ctx.compose(|ctx| {
+            table::hooks::padding(OUTER_PADDING, |wh, ctx| {
+                ctx.add(scroll_view::AutoScrollViewWithCtx {
+                    xy: Xy::zero(),
+                    scroll_bar_width: 4.px(),
+                    height: wh.height,
+                    content: |ctx| {
+                        table::hooks::vertical(images.chunks(*max_items_per_row).map(|images| {
+                            table::hooks::fixed(THUMBNAIL_WH.height, {
+                                table::hooks::horizontal(images.iter().map(|image| {
+                                    render_thumbnail(image, project_id, &on_internal_event)
+                                }))
+                            })
+                        }))(wh, ctx)
+                    },
+                });
+            })(wh, ctx)
+        });
+
+        let background = simple_rect(wh, color::STROKE_NORMAL, 1.px(), color::BACKGROUND)
+            .attach_event(|event| {
+                if let namui::Event::MouseDown { event } = event {
+                    event.stop_propagation();
+                    if !event.is_local_xy_in() {
+                        on_event(Event::Close);
+                    }
```

Comment:
Why it closes the picker?

Reply:
Clicking outside of picker means `User wants to do something other than adding image`.
So if user clicks outside, picker should be closed.

Reply:
Then all windows on right side will be closed when click left side?

Reply:
So, it's the problem that 'adding cg' and 'adding image' is showing on right panel which user don't expect it is temporary input panel, unlike modal.

Reply:
Then we should use modal design for pickers. right?

Reply:
> Then we should use modal design for pickers. right?

If it is right, I think it would be good to merge this pr and create a new issue.

Reply:
modal isn't the only option, i guess. it should be considered in other options after studying other app's cases.